packages/application-shell/src/components/application-shell/redirect-to-logout.tsx

```diff
@@ -1,22 +1,42 @@
-import type { RouteProps } from 'react-router-dom';
+import type { ApplicationWindow } from '@commercetools-frontend/constants';
 
 import React from 'react';
+import { useLocation } from 'react-router-dom';
 import { LOGOUT_REASONS } from '@commercetools-frontend/constants';
 import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import { STORAGE_KEYS } from '../../constants';
 import Redirector from '../redirector';
+import RedirectToLogin from './redirect-to-login';
+
+declare let window: ApplicationWindow;
+
+type Props = {
+  reason?: typeof LOGOUT_REASONS[keyof typeof LOGOUT_REASONS];
+};
 
 // When the application redirects to this route, we always force a hard redirect
 // to the logout route of the authentication service.
-const RedirectToLogout = (props: RouteProps) => {
+const RedirectToLogout = (props: Props) => {
+  const location = useLocation();
   const servedByProxy = useApplicationContext(
     (context) => context.environment.servedByProxy
   );
+
+  if (window.app.__DEVELOPMENT__) {
+    // Remove the `sessionToken` from storage, so that the AppShell can initiate
+    // a new authorization flow.
+    window.sessionStorage.removeItem(STORAGE_KEYS.SESSION_TOKEN);
+    window.sessionStorage.removeItem(STORAGE_KEYS.SESSION_SCOPE);
+    window.sessionStorage.removeItem(STORAGE_KEYS.IS_AUTHENTICATED);
```

Comment:
Nit: Maybe we can have a small `development-oidc-localstorage.ts` module which allows setting and unsetting things. Just to not spread knowledge o the keys into different parts of the shell.

Reply:
Thanks, I'll see what I can do.

Reply:
https://github.com/commercetools/merchant-center-application-kit/pull/1934/commits/05a7dd9668a0285e0a4fef80e637c6be1c4cf0c3